src/app.js

```diff
@@ -1,35 +1,47 @@
 import React from "react";
+import {Provider} from 'react-redux';
+import {startMovieRequest} from './actions/movie';
 import ReactDOM from "react-dom";
+import configureStore from './store/configureStore';
 import {ErrorBoundary} from './components/ErrorBoundary';
 import SearchPage from './pages/SearchPage';
 import DetailsPage from './pages/DetailsPage';
 
 import 'normalize.css/normalize.css';
 import './styles.scss';
 
+const store = configureStore();
+
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     // For shocasing purposes
     this.state = {
-      page: 'details'
+      page: 'search'
     };
   }
 
+  handleMovieSelected = (movieId) => {
+    store.dispatch(startMovieRequest(movieId)).then(() => {
+      this.setState({selectedMovieId: movieId, page: 'details'});
+    });
+  }
+
   onPageChange = (page) => {
     this.setState({page});
   }
 
   render() {
     return (
-      <div className="App">
-        <ErrorBoundary>
-          {this.state.page === 'search'? 
-            <SearchPage onPageChange={this.onPageChange}/>: 
-            <DetailsPage onPageChange={this.onPageChange}/>}
-        </ErrorBoundary>
-      </div>
+      <Provider store = {store}>
+        <div className="App">
+          <ErrorBoundary>
+            {this.state.page === 'search' && <SearchPage handleMovieSelected={this.handleMovieSelected}/>}
```

Comment:
Another important aspect of Redux is to avoid passing props to many levels of components.

As far you have actions that can change your state, you can map your components/containers to receive those properties directly instead of passing it from father to child component.

For example in your SearchPage you could just render it because SearchPage could receive this handleMovieSelected directly as an action that will change your state and dispatch your request.

Is not a good practice to use store.dispatch directly.